fix: does not comply with psr-4 autoloading standard

gerardroche opened the pull request:
Fixes PSR-4 autoloading issues as seen when installing via:

```
composer install --no-dev -o
```

These are the warnings that this PR fixes:

```
Class Phpactor\ClassMover\Tests\Adapter\TolerantParser\ExampleTrait located in ./lib/ClassMover/Tests/Adapter/TolerantParser/examples/Example6.php does not comply with psr-4 autoloading standard. Skipping.
Class Phpactor\ClassMover\Tests\Adapter\TolerantParser\Example7 located in ./lib/ClassMover/Tests/Adapter/TolerantParser/examples/Example7.php does not comply with psr-4 autoloading standard. Skipping.
Class Phpactor\ClassMover\Tests\Adapter\TolerantParser\Example5Interface located in ./lib/ClassMover/Tests/Adapter/TolerantParser/examples/Example5.php does not comply with psr-4 autoloading standard. Skipping.
Class Phpactor\Extension\LanguageServerPsalm\Model\Excepteion\PsalmProcessError located in ./lib/Extension/LanguageServerPsalm/Model/Excepteion/PhpstanProcessError.php does not comply with psr-4 autoloading standard. Skipping.
Class Phpactor\CodeTransform\Tests\Adapter\WorseReflection\Refactor\WorseGenerateMethodTest located in ./lib/CodeTransform/Tests/Adapter/WorseReflection/Refactor/WorseGenerateMemberTest.php does not comply with psr-4 autoloading standard. Skipping.
```

dantleech commented:
Thanks - I think for the test files it would be better to add a `.gitattributes` file:

```
*.php diff=php
/path1/to/tests/to/ignore export-ignore
```

that will prevent them from being exported. but otherwise if we can't get composer to ignore them easily happy to make the change you suggest here. 

the "PslamProcessError" file is not used and can be deleted AFAICS

gerardroche commented:
I removed two unused exceptions `PslamProcessError` and `PhanProcessError`. There is no reference to them anywhere, and there was a typo in the namespace "Excepteion".

I added gitattributes export rules and baseline line endings rule. I wasn't sure what files can be ignored,  so I just added some obvious ones. You will know better than me which files should be ignored.

The export-ignore rules are good, but they don't help for my use case because I install phpactor directly from a clone. 

I noticed that there are tests nested inside `lib/*/Tests/`. All of those files get populated into the autoload classmap  generated by composer. The `tests/` files don't when you do `composer --no-dev` because those are declared in the composer `autoload-dev`. 

I think composer has ways to handle those `lib/*/Tests/` files, but I don't know off the top of my head. I don't think it's a major deal. It might save some memory, but performance-wise it should be negligible.

dantleech commented:
Great, thanks!
